app/public/templates/scene.html

```diff
@@ -99,7 +99,7 @@
           class="link"
           hint-button='jsonData: {"newHintImg": "/static/img/lightOn.png", "hintImg": "/static/img/lightOff.png", "width": 0.08, "height": 0.08}'
           animation__onclick="property: position; dur: 0; startEvents: click"
-          animation-on-click-run='blackboardData: {"jsonData": {"text": {% raw json.dumps(scene_dict["hints"]) %}, "currPosition": 0}, "componentType": "text-pane"}'
+          animation-on-click-run='blackboardData: {"jsonData": {"data": {% raw json.dumps( scene_dict["hints"] ) %}, "currPosition": 0}, "componentType": "text-pane"}'
         >
```

Comment:
Because of the schema changes that were made to the text-pane component, the hints pane isn't working at the moment since the jsonData is not following the correct schema. I was confused about the `{% raw json.dumps(scene_dict["hints"]) %}` part - is the `{% %}` a template engine system based on jinja?

Old Schema:
`{"text": ["hi", "hello"], "currPosition":0}`

New Schema:
`{"data": [{"text": "hi"}, {"text": "hello", "imageSrc": "path"}], "currPosition": 0}`

Reply:
Ah yes, `{% %}` is part of the tornado template system. Everything inside of it is _basically_ Python. 
So, what we were doing before is using the json library to dump the array in JS friendly manner (hence `raw`).
I fixed it by changing it to:
`{% raw json.dumps( [ {"text": x} for x in scene_dict["hints"] ] ) %}`.
List comprehension in Python,
so now `["generic hint for all scenes", "scene specific hint 1"]` will become `[{'text': 'generic hint for all scenes'}, {'text': 'scene specific hint 1'}]` and then we json.dumps it.
Overall this will give what the text-pane now expects!